### ISSUE TITLE
Support "indent like" configuration

### ISSUE THREAD

**Issue by robhanlon22** [greglook/cljstyle#60]
clj-kondo has a nice feature where you can configure it to lint a particular macro as if it were a different macro, so you don't have to write custom linters for macros if they're not necessary. That got me thinking—it would be really nice if cljstyle supported something like that in the `:indents` configuration, e.g.

```clojure
;; .cljstyle
{:indents {def-special-fn defn}}
```

Would indent `def-special-fn` like `defn`. It might even be worth adding a new configuration key so you could opt into certain special formats, as the example above wouldn't format its arg vector liked defn:

**Comment by greglook** [comment]
Yup! I have been playing with a similar idea; this and a few other issues have led me to want to revamp how `cljstyle` config is structured to make it more rule-focused instead of a large flat map. Unfortunately I haven't had a lot of spare time to work on this lately (puppies are hard 😭) but I'm hoping to get it done this month.

**Comment by robhanlon22** [comment]
> Yup! I have been playing with a similar idea; this and a few other issues have led me to want to revamp how `cljstyle` config is structured to make it more rule-focused instead of a large flat map. Unfortunately I haven't had a lot of spare time to work on this lately (puppies are hard 😭) but I'm hoping to get it done this month.

PUPPIES
